ruby/spec/gilded_rose_spec.rb

```diff
@@ -9,12 +9,78 @@
       expect(items[0].name).to eq "foo"
     end
 
-    it "degrades quality over time by one if not at sell by date" do
-      items = [Item.new("foo", 10, 50)]
+    it "degrades quality over time by one if not at sell in date" do
+      items = [Item.new("foo", 10, 30)]
       GildedRose.new(items).update_quality()
-      expect(items[0].quality).to eq 49
+      expect(items[0].quality).to eq 29
+    end
+
+    it "decreases sell in date by one day" do
+      items = [Item.new("foo", 10, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].sell_in).to eq 9
+    end
+
+    it "degrades quality twice as fast if past the sell in date" do
+      items = [Item.new("foo", 0, 50)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 48
+    end
+ 
+    it "never allows quality to be negative" do
+      items = [Item.new("foo", 3, 0)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 0
+    end
+  
+    it "increases quality over time if name is Aged Brie" do
+      items = [Item.new("Aged Brie", 3, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 31
+    end 
+ 
+    it "never allows quality to increase over 50" do
+      items = [Item.new("Aged Brie", 3, 50)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 50
+    end
+  
+    it "does not need to decrease qulaity for Sulfuras" do
+      items = [Item.new("Sulfuras, Hand of Ragnaros", 3, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 30
+    end
+   
+    it "does not need to sell Sulfuras" do
+      items = [Item.new("Sulfuras, Hand of Ragnaros", 0, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 30
     end
     
+    it "allows back stage passes by one more than 10 days before the concert" do
```

Comment:
Missing some words here.  `to increase`?  Take a look at your test descriptions with fresh eyes in general, as long as you'll be making some changes.